control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go

```diff
@@ -1857,6 +1858,17 @@ func (r *reconciler) ensureCloudResourcesDestroyed(ctx context.Context, hcp *hyp
 		log.Info("Persistent volumes are removed")
 	}
 
+	log.Info("Ensuring volumn snapshots are removed")
+	removed, err = r.ensureVolumeSnapshotsRemoved(ctx)
```

Comment:
I would pass the HCP Namespace by argument and put the `ensureVolumeSnapshotsRemoved` as a public function in a more common place to be reused (E.G support/utils (?)).

Reply:
> I would pass the HCP Namespace by argument 

This is cleaning up volume snapshots within the guest cluster. The hcp namespace would not be relevant here. 

> put the ensureVolumeSnapshotsRemoved as a public function in a more common place to be reused (E.G support/utils (?)).

That would be inconsistent with the current way this code is structured. I'm following the pattern used by this function for other resources that need to be cleaned up. 

examples a few lines up. 

```
	removed, err = r.ensureIngressControllersRemoved(ctx, hcp)
	if err != nil {
		errs = append(errs, err)
	}
	if !removed {
		remaining.Insert("ingress-controllers")
	} else {
		log.Info("Ingress controllers are removed")
	}
	log.Info("Ensuring load balancers are removed")
	removed, err = r.ensureServiceLoadBalancersRemoved(ctx)
	if err != nil {
		errs = append(errs, err)
	}
	if !removed {
		remaining.Insert("loadbalancers")
	} else {
		log.Info("Load balancers are removed")
	}
	log.Info("Ensuring persistent volumes are removed")
	removed, err = r.ensurePersistentVolumesRemoved(ctx)
```

Reply:
@jparrill any more thoughts on this PR?